eclipse-collections/src/main/java/org/eclipse/collections/impl/AbstractRichIterable.java

```diff
@@ -680,6 +681,18 @@ public boolean containsAll(Collection<?> collection)
         return target;
     }
 
+    /**
+     * This method will count the number of occurrences of each value calculated by applying the
+     * function to each element of the collection.
+     *
+     * @since 8.2
+     */
+    @Override
+    public <V> Bag<V> countBy(Function<? super T, ? extends V> function)
+    {
+        return this.countBy(function, Bags.mutable.empty());
```

Comment:
I can't reference Bags.mutable.empty() in the interface, but thought this is a better abstract implementation than I can provide in the interface.